plugins/opener/src/scope.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use std::{marker::PhantomData, path::PathBuf, sync::Arc};
+
+use tauri::{ipc::ScopeObject, utils::acl::Value, AppHandle, Manager, Runtime};
+
+use url::Url;
+
+use crate::{scope_entry::EntryRaw, Error};
+
+#[derive(Debug)]
+pub enum Entry {
+    Url(glob::Pattern),
+    Path(Option<PathBuf>),
+}
+
+impl ScopeObject for Entry {
+    type Error = Error;
+
+    fn deserialize<R: Runtime>(
+        app: &AppHandle<R>,
+        raw: Value,
+    ) -> std::result::Result<Self, Self::Error> {
+        serde_json::from_value(raw.into())
+            .and_then(|raw| {
+                let entry = match raw {
+                    EntryRaw::Url { url } => Entry::Url(
+                        glob::Pattern::new(&url)
+                            .map_err(|e| serde::de::Error::custom(e.to_string()))?,
+                    ),
+                    EntryRaw::Path { path } => {
+                        let path = match app.path().parse(path) {
+                            Ok(path) => Some(path),
+                            #[cfg(not(target_os = "android"))]
+                            Err(tauri::Error::UnknownPath) => None,
+                            Err(err) => return Err(serde::de::Error::custom(err.to_string())),
+                        };
+
+                        Entry::Path(path)
+                    }
+                };
+
+                Ok(entry)
+            })
+            .map_err(Into::into)
+    }
+}
+
+#[derive(Debug)]
+pub struct Scope<'a, R: Runtime, M: Manager<R>> {
+    allowed: Vec<&'a Arc<Entry>>,
+    denied: Vec<&'a Arc<Entry>>,
+    manager: &'a M,
+    _marker: PhantomData<R>,
+}
+
+impl<'a, R: Runtime, M: Manager<R>> Scope<'a, R, M> {
+    pub(crate) fn new(
+        manager: &'a M,
+        allowed: Vec<&'a Arc<Entry>>,
+        denied: Vec<&'a Arc<Entry>>,
+    ) -> Self {
+        Self {
+            manager,
+            allowed,
+            denied,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn is_allowed(&self, path_or_url: &str) -> crate::Result<bool> {
+        let url = Url::parse(path_or_url).ok();
```

Comment:
This will not work, Windows paths like `C:/Users/Tony/.gitconfig` will be parsed as an URL